MultiBuild/BlueprintManager.cs

```diff
@@ -736,81 +649,201 @@ public static List<BuildPreview> Paste(Vector3 targetPos, float yaw, bool pasteI
-                bp.outputOffset = positionData.insertOffset;
-                bp.outputToSlot = positionData.endSlot;
-                bp.inputFromSlot = positionData.startSlot;
-                bp.condition = positionData.condition;
+            bp.lrot = positionData.absoluteInserterRot;
+            bp.lrot2 = positionData.absoluteInserterRot2;
 
-                bp.filterId = copiedInserter.filterId;
+            bp.inputToSlot = 1;
+            bp.outputFromSlot = 0;
 
-                if (pastedEntities.TryGetValue(positionData.inputOriginalId, out PastedEntity inputEntity))
-                {
-                    bp.input = inputEntity.buildPreview;
-                }
-                else
-                {
-                    bp.inputObjId = positionData.inputObjId;
-                }
+            bp.inputOffset = positionData.pickOffset;
+            bp.outputOffset = positionData.insertOffset;
+            bp.outputToSlot = positionData.endSlot;
+            bp.inputFromSlot = positionData.startSlot;
+            bp.condition = positionData.condition;
 
-                if (pastedEntities.TryGetValue(positionData.outputOriginalId, out PastedEntity outputEntity))
-                {
-                    bp.output = outputEntity.buildPreview;
-                }
-                else
-                {
-                    bp.outputObjId = positionData.outputObjId;
-                }
+            bp.filterId = inserter.filterId;
+
+            if (pastedEntities.TryGetValue(positionData.inputOriginalId, out PastedEntity inputEntity))
+            {
+                bp.input = inputEntity.buildPreview;
+            }
+            else
+            {
+                bp.inputObjId = positionData.inputObjId;
+            }
 
-                pastedEntities.Add(copiedInserter.originalId, new PastedEntity()
+            if (pastedEntities.TryGetValue(positionData.outputOriginalId, out PastedEntity outputEntity))
+            {
+                bp.output = outputEntity.buildPreview;
+            }
+            else
+            {
+                bp.outputObjId = positionData.outputObjId;
+            }
+
+            lock (pastedEntities)
+            {
+                pastedEntities.Add(inserter.originalId, new PastedEntity()
                 {
-                    buildPreview = bp
+                    type = EPastedType.INSERTER,
+                    index = index,
+                    buildPreview = bp,
                 });
-
+            }
+            lock (previews)
+            {
                 previews.Add(bp);
             }
+        }
 
-            lastMaxWidth = currentMaxWidth;
+        public static void ActivateColliders( Vector3 center)
+        {
+            NearColliderLogic nearCdLogic = GameMain.data.mainPlayer.controller.actionBuild.nearcdLogic;
 
-            return previews;
+            nearCdLogic.activeColHashCount = 0;
+            nearCdLogic.MarkActivePos(center);
+
+            var sqrAreaRadius = 4f * 4f;
```

Comment:
Let the compiler do the heavy lifting here :D

```suggestion
            const float sqrAreaRadius = 4f * 4f;
```

Reply:
thanks